shared/js/dialer/keypad.js

```diff
@@ -122,6 +124,8 @@ var KeypadManager = {
     this._onCall = !!oncall;
 
     this.phoneNumberView.value = '';
+    var updateCaretPosition = this._updateCaretPosition.bind(this);
+    this.phoneNumberView.addEventListener('click', updateCaretPosition);
```

Comment:
Since ``updateCaretPosition`` is used only once you can directly assign the bound function to the listener:

 ```javascript
this.phoneNumberView.addEventListener('click', this._updateCaretPosition.bind(this));
```

I'm not super-fond of bound listeners since they tend to leak when used in tests but we already have some in this file so it's OK for this patch.